e2e-test-server/app.go

```diff
@@ -0,0 +1,52 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"context"
+	"log"
+	"os"
+	"os/signal"
+	"time"
+
+	"github.com/GoogleCloudPlatform/opentelemetry-operations-go/e2e-test-server/endtoendserver"
+)
+
+func main() {
+	runCtx, cancelRunning := context.WithCancel(context.Background())
+
+	// Gracefully close on SIGINT
+	interrupted := make(chan os.Signal, 1)
+	signal.Notify(interrupted, os.Interrupt)
+	go func() {
+		<-interrupted
+		log.Println("Received interrupt signal, shutting down.")
+		cancelRunning()
+	}()
+
+	server, err := endtoendserver.New()
+	if err != nil {
+		log.Fatalf("Could not initialize server: %v", err)
+	}
+
+	err = server.Run(runCtx)
+	if err != nil {
+		log.Printf("Shutting down due to unexpected error: %v", err)
+	}
+
+	shutdownCtx, cancelShutdown := context.WithTimeout(context.Background(), 10*time.Second)
```

Comment:
Here server will shutdown anyway, isn't it better to separate this log?!
```suggestion
	err = server.Run(runCtx)
	if err != nil {
		log.Printf("Unexpected error occurred: %v", err)
	}
	
	log.Print("Shutting down")
	shutdownCtx, cancelShutdown := context.WithTimeout(context.Background(), 10*time.Second)
```